src/frontend/flight/handler.rs

```diff
@@ -171,22 +174,24 @@ impl SeafowlFlightHandler {
 
         debug!("Processing data change with {num_rows} rows for url {url}");
         match tokio::time::timeout(
-            Duration::from_secs(self.context.config.misc.sync_conf.sync_lock_timeout_s),
-            self.sync_manager.write(),
+            Duration::from_secs(self.context.config.misc.sync_conf.write_lock_timeout_s),
+            self.sync_writer.write(),
         )
         .await
         {
-            Ok(mut sync_manager) => {
-                let (mem_seq, dur_seq) = sync_manager
-                    .enqueue_sync(
-                        log_store,
-                        cmd.sequence_number,
-                        cmd.origin,
-                        sync_schema.expect("Schema available"),
-                        cmd.last,
-                        batches,
-                    )
-                    .await?;
+            Ok(mut sync_writer) => {
+                sync_writer.enqueue_sync(
+                    log_store,
+                    cmd.sequence_number,
+                    cmd.origin,
+                    sync_schema.expect("Schema available"),
+                    cmd.last,
+                    batches,
+                )?;
+
+                sync_writer.flush().await?;
```

Comment:
It's still the case that this `flush()` op won't return until we've had a chance to see if we need to flush and wait for the flush to finish, right? (i.e. Seafowl will never end up using more memory than the limit)

Reply:
The `flush` call there checks the criteria and while it is met it flushes table after table. The memory usage can only go down inside/after the call.